app/src/main/java/app/f3d/F3D/android/MainActivity.java

```diff
@@ -4,67 +4,73 @@
 import android.net.Uri;
 import android.os.Bundle;
 import android.util.Log;
-import android.app.Activity;
 
+import androidx.activity.result.ActivityResultLauncher;
+import androidx.activity.result.contract.ActivityResultContracts;
+import androidx.appcompat.app.AppCompatActivity;
 import androidx.constraintlayout.widget.ConstraintLayout;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
 
-import java.io.FileNotFoundException;
-import java.io.IOException;
-import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Objects;
 
-public class MainActivity extends Activity {
+import app.f3d.F3D.android.Utils.FileInteractionContract;
+import app.f3d.F3D.android.Utils.FileType;
+import app.f3d.F3D.android.Utils.FileUtils;
 
-    MainView mView;
+public class MainActivity extends AppCompatActivity {
+
+    private MainView mView;
+    private ActivityResultLauncher<Void> fileInteractionLauncher;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
 
         super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        ConstraintLayout mainLayout = findViewById(R.id.mainLayout);
+
+        FloatingActionButton addButton = findViewById(R.id.addButton);
+
+        Objects.requireNonNull(getSupportActionBar()).hide();
```

Comment:
what does this line do?

Reply:
This is the Actionbar which im hiding, the purple bar on top

The FAB is the button to select files

Reply:
Right, I was just confused about `Objects.requireNonNull` but I guess it's just a wrapper that checks if the object is null before calling `hide()`.